rest-api/tools/setup_database.sh

```diff
@@ -100,36 +91,56 @@ function finish {
 }
 trap finish EXIT
 
-echo '{"db_connection_string": "'$CONNECTION_STRING'", ' \
-     ' "backup_db_connection_string": "'$BACKUP_CONNECTION_STRING'", '\
-     ' "rdr_db_password": "'$RDR_PASSWORD'", ' \
-     ' "root_db_password": "'$ROOT_PASSWORD'", ' \
-     ' "read_only_db_password": "'$READONLY_PASSWORD'", ' \
-     ' "db_connection_name": "'$INSTANCE_CONNECTION_NAME'", '\
-     ' "backup_db_connection_name": "'$BACKUP_INSTANCE_NAME'", '\
-     ' "db_user": "'$RDR_DB_USER'", '\
-     ' "db_name": "'$DB_NAME'" }' > $TMP_DB_INFO_FILE
-
-for db_name in "rdr" "metrics"; do
-  if [ "${UPDATE_PASSWORDS}" = "Y" ]
-  then
-    cat tools/update_passwords.sql | envsubst > $UPDATE_DB_FILE
-  else
-    cat tools/create_db.sql | envsubst > $UPDATE_DB_FILE
-  fi
-
-  run_cloud_sql_proxy
+function get_existing_db_config {
+  "Getting database info..."
+  tools/install_config.sh --key db_config --instance $INSTANCE \
+   --creds_file ${CREDS_FILE} --config_output "$TMP_DB_INFO_FILE"
+}
 
-  if [ "${UPDATE_PASSWORDS}" = "Y" ]
+if [ "${UPDATE_PASSWORDS}" = "Y" ] || [ "${CREATE_INSTANCE}" == "Y" ]
   then
-    echo "Updating database user passwords..."
+    	echo "Updating database user passwords..."
+	randpw
+	ROOT_PASSWORD=$new_password
+	randpw
+	RDR_PASSWORD=$new_password
+	randpw
+	READONLY_PASSWORD=$new_password
+
+	echo '{"db_connection_string": "'$CONNECTION_STRING'", ' \
+	     ' "backup_db_connection_string": "'$BACKUP_CONNECTION_STRING'", '\
+	     ' "rdr_db_password": "'$RDR_PASSWORD'", ' \
+	     ' "root_db_password": "'$ROOT_PASSWORD'", ' \
+	     ' "read_only_db_password": "'$READONLY_PASSWORD'", ' \
+	     ' "db_connection_name": "'$INSTANCE_CONNECTION_NAME'", '\
+	     ' "backup_db_connection_name": "'$BACKUP_INSTANCE_NAME'", '\
+	     ' "db_user": "'$RDR_DB_USER'", '\
+	     ' "db_name": "'$DB_NAME'" }' > $TMP_DB_INFO_FILE
+
+	echo "Setting root password..."
+	#gcloud sql instances set-root-password $INSTANCE_NAME --password $ROOT_PASSWORD
+	gcloud sql users set-password root % --instance $INSTANCE_NAME --password $ROOT_PASSWORD
+
+	if [ "${UPDATE_PASSWORDS}" = "Y" ]
+	    then 
+		echo "updating passwords for database"
+		cat tools/update_passwords.sql | envsubst > $UPDATE_DB_FILE
+	fi
+	if [ "${CREATE_INSTANCE}" = "Y" ]
+	    then
+		echo "creating new database"
+		for db_name in "rdr" "metrics"; do
+	       	   cat tools/create_db.sql | envsubst > $UPDATE_DB_FILE
```

Comment:
db_name is set to 'rdr' and 'metrics' respectively as it iterates. so isn't that what we're doing here already?

Reply:
In bash variable names are case sensitive... db_name != DB_NAME.

It could be that if you did "for DB_NAME" it would work. It might be better to assign DB_NAME to db_name here.

Reply:
Sorry, I'm an idiot. :) This should work.

Reply:
😄 done